src/zenml/new/pipelines/pipeline.py

```diff
@@ -635,6 +635,7 @@ def _run(
                 )
 
             stack = Client().active_stack
+            stack.validate()
```

Comment:
This validation in already happening in `stack.prepare_pipeline_deployment` later in this method. I guess you want it here so we make sure the image builder doesn't break right? Can we somehow verify that at stack registration/update? I guess not as it won't be called when registering from the dashboard at least?

Reply:
Yeah this is technically redundant but I added the call here to ensure we'll always validate before trying to run any pipelines on a stack